pallets/rc-migrator/src/accounts.rs

```diff
@@ -775,29 +806,287 @@ impl<T: Config> AccountsMigrator<T> {
+		}
+
+		for (channel_id, info) in hrmp::HrmpOpenChannelRequests::<T>::iter() {
+			let sender: T::AccountId = channel_id.sender.into_account_truncating();
+			let sender_deposit = info.sender_deposit;
+			if sender_deposit > 0 {
+				rc_reserves
+					.entry(sender.clone())
+					.and_modify(|r: &mut u128| *r = r.saturating_add(sender_deposit))
+					.or_insert(sender_deposit);
+			}
+		}
+
+		for (_, info) in Paras::<T>::iter() {
+			let manager = info.manager;
+			let manager_deposit = info.deposit;
+			if manager_deposit > 0 {
+				rc_reserves
+					.entry(manager.clone())
+					.and_modify(|r: &mut u128| *r = r.saturating_add(manager_deposit))
+					.or_insert(manager_deposit);
+			}
+		}
+
+		for (who, mut reserved_kept) in rc_reserves {
+			// Holds migration is prioritized over keeping unnamed reserves on the relay chain
+			let total_reserved = <T as Config>::Currency::reserved_balance(&who);
+			let total_hold = pallet_balances::Holds::<T>::get(&who)
+				.into_iter()
+				.map(|h| h.amount)
+				.sum::<T::Balance>();
+			reserved_kept = reserved_kept.min(total_reserved.saturating_sub(total_hold));
+			if reserved_kept == 0 {
+				continue;
+			}
+
+			// We need to keep some reserved balance (reserved_kept) on the relay chain, together
+			// with the Relay Chain existential deposit. Notice that this cannot be greater than
+			// the total balance of the account.
+			let ed = <T as Config>::Currency::minimum_balance();
+			let total_balance = <T as Config>::Currency::total_balance(&who);
+			let free_kept = ed.min(total_balance.saturating_sub(reserved_kept));
+			rc_reserved_kept.insert(who.clone(), reserved_kept);
+			rc_free_kept.insert(who.clone(), free_kept);
+		}
+		Self { rc_reserved_kept, rc_free_kept }
+	}
+}
+
 #[cfg(feature = "std")]
-impl<T: Config> crate::types::RcMigrationCheck for AccountsMigrator<T> {
-	// rc_total_issuance_before
-	type RcPrePayload = BalanceOf<T>;
+impl<T> AccountsMigrationChecker<T> {
+	// Translate the lock id to a u8 for both RC and AH using the different encodings for easier
+	// comparison. Lock Ids type is  `LockIdentifier = [u8; 8]` on Polkadot.
+	pub fn lock_id_to_u8(lock_id: [u8; 8], _chain_type: ChainType) -> u8 {
+		match lock_id.as_slice() {
+			b"staking " => 0,
+			b"vesting " => 1,
+			b"pyconvot" => 2,
+			_ => {
+				log::error!(
+					target: LOG_TARGET,
+					"Unknown lock id: {:?}",
+					lock_id
+				);
+				3
+			},
+		}
+	}
+
+	// Translate the freeze id to a u8 for both RC and AH using the different encodings for easier
+	// comparison. Freeze Ids type is enum `RuntimeFreezeReason` on Polkadot. Here we use the
+	// encoded enum.
+	pub fn freeze_id_encoding_to_u8(freeze_id: Vec<u8>, chain_type: ChainType) -> u8 {
+		let nomination_pools_encoding = match chain_type {
+			ChainType::RC => [39, 0],
+			ChainType::AH => [80, 0],
+		};
+		if freeze_id.as_slice() == nomination_pools_encoding {
+			0
+		} else {
+			log::error!(target: LOG_TARGET, "Unknown freeze id: {:?}", freeze_id);
```

Comment:
```suggestion
			panic!("Unknown freeze id: {:?}", freeze_id);
```
I think this is some serious error when it is unknown.

Reply:
Yes, changed to `panic` in https://github.com/polkadot-fellows/runtimes/pull/674/commits/4bbb3ea51b858323e9c77d7443e49b848684bb26